### DISCUSSION TITLE
How is textual compared to other TUI libraries?

### DISCUSSION THREAD

**Discussion by ws1088** [Textualize/textual#341] [Q&A]
Hi, I am researching on what TUI libraries are out there. What's the advantage of `textual` over other TUI libraries out there? Also what are some of the use cases which TUI applications are desired? Thanks in advance.

**Comment by USLTD** [comment]
Hello @ws1088 .
Advantages of [**textual**](https://github.com/Textualize/textual) over other <kbd>TUI (Text-based User Interface)</kbd> libraries are following:
```diff
@@ Asynchronous, which is both advantage and disadvantage depending on situation @@
+ Supports `CSS` for styling
+ Based on `rich` meaning that it can render anything that `rich` can which means beautiful panels, trees, progress bars and more
+ Easy to design
+ Easy to add/create new widgets
- Lack of documentation (by @jeffwright13)
@@ Hard to debug possibly (by @jeffwright13) @@
- Speed (not versus libraries implemented in Python)
```

**Reply by jeffwright13** [reply]
> Hello @ws1088 . Advantages of [**textual**](https://github.com/Textualize/textual) over other TUI (Text-based User Interface) libraries are following:
> 
> ```diff
> @@ Asynchronous, which is both advantage and disadvantage depending on situation @@
> + Supports `CSS` for styling
> + Based on `rich` meaning that it can render anything that `rich` can which means beautiful panels, trees, progress bars and more
> + Easy to design
> + Easy to add/create new widgets
> - Speed (not versus libraries implemented in Python)
> ```

I'd say those are accurate. The only things I would add as a negative right now would be:
- lack of documentation
- difficulty of debugging (although this seems like an anycio thing, not specifically Textual)

**Reply by willmcgugan** [reply]
Textual is still a very early release. Docs and debugging tools are coming.

**Reply by jeffwright13** [reply]
> Textual is still a very early release. Docs and debugging tools are coming.

Yep, thanks! But even at this stage it's too compelling to not mess around with. :-)
